src/main/java/com/microsoft/spring/data/gremlin/conversion/MappingGremlinConverter.java

```diff
@@ -86,11 +87,14 @@ private Object getFieldValue(@NonNull Object domain, @NonNull String fieldName)
         final ConvertingPropertyAccessor accessor = this.getPropertyAccessor(domain);
         final GremlinPersistentEntity<?> persistentEntity = this.getPersistentEntity(domain.getClass());
         final PersistentProperty property = persistentEntity.getPersistentProperty(fieldName);
-        Assert.notNull(property, "persistence property should not be null");
+        if (!fieldName.equals(Constants.PROPERTY_ID)) {
```

Comment:
should not workaround the PROPERTY_ID, as the field which has `@Id` without id name. And also if you want to support Id optional, this may not the right place from the method name.